### ISSUE TITLE
Problems with POT Quantization

### ISSUE THREAD

**Issue by glucasol** [openvinotoolkit/openvino#17389]
Hi everyone,

I am having some problems to quantize my OpenVINO model with POT.

I have an anomaly detection model trained using this [Anomalib repo](https://github.com/openvinotoolkit/anomalib). I converted the trained model to OpenVINO format (.bin and .xml) with Model Optimizer MO. 

The structure of the dataset used to train/test the model is:

<ul>
  <li>Dataset Folder
  <ul>
  <li>Train
    <ul>
      <li>All normal images</li>
    </ul>
  </li>
  <li>Test
    <ul>
      <li>Normal <ul> <li>All good images</li> </ul></li>
      <li>Abnormal<ul> <li>All anomalous images</li> </ul></li>
    </ul>
  </li>
  <li>Masks
    <ul>
     <li>Abnormal<ul> <li>All anomalous masks</li> </ul></li>
    </ul>
  </li>
  </ul>
  </li>
</ul>

I want to run this POT command (As mentioned here: [POT](https://docs.openvino.ai/latest/pot_compression_cli_README.html) ):   `pot -c <path_to_config_file>`

I have created the .json configuration file like this [example](https://docs.openvino.ai/latest/pot_configs_examples_README.html)

But I need to have a config .yaml file, with is the Accuracy Checker configuration file.

So following this [Accuracy Checker Guide](https://docs.openvino.ai/latest/omz_tools_accuracy_checker.html#doxid-omz-tools-accuracy-checker), I need to create a config.yaml file similiar as below:
````
models:
- name: model_name
  launchers:
  - framework: openvino
    device: CPU
    model: path_to_model/alexnet.xml
    weights: path_to_weights/alexnet.bin
    adapter: anomaly_segmentation
  datasets:
    - name: dataset_name
      annotation: annotation.pickle
      data_source: images_folder

      preprocessing:
      - type: resize
        dst_width: 256
        dst_height: 256

      - type: normalization
        mean: imagenet

      - type: crop
        dst_width: 227
        dst_height: 227

     metrics:
     - type: accuracy
````

**After following all  these steps, I came up with some doubts:**
First, in launchers section, with adapter should I use, since my model is an anomaly detection model?
Second, in datasets section, what do I need to specify in annotation subsection, since I do not have an annotation file? [Anomalib](https://github.com/openvinotoolkit/anomalib) do not need annotation file, since the train dataset only has normal images and the test dataset is separated in normal and anomalous folders.

Can You help me with these doubts?

Thanks!

**Comment by Iffa-Meah** [comment]
@glucasol If you don't have annotations in your dataset, it is recommended to apply model quantization with the[ Default Quantization method](https://docs.openvino.ai/latest/pot_default_quantization_usage.html#doxid-pot-default-quantization-usage) without accuracy control, using an unannotated dataset.

Unless you have a specific reason to use  [Quantized Model with Accuracy Control](https://docs.openvino.ai/latest/pot_accuracyaware_usage.html)

You need to create a Python script using an API of Post-Training Optimization Tool (POT) and implement data preparation logic and quantization pipeline.

There are 3 crucial steps:
1.Prepare data and dataset interface.
2.Select quantization parameters.
3.Define and run quantization process.

In step 1, when defining DataLoader, the annotation is set to None.
You may refer to the scripts implementation section in  [this guide](https://docs.openvino.ai/latest/pot_default_quantization_usage.html#prepare-data-and-dataset-interface).

**Comment by alexsu52** [comment]
Hi @glucasol, OpenVINO POT will be deprecated in the next OpenVINO release https://github.com/openvinotoolkit/openvino/pull/16758 and the new Python post-training quantization API from the [NNCF](https://github.com/openvinotoolkit/nncf) will be the main way to quantize the models.

NNCF has the [example](https://github.com/openvinotoolkit/nncf/tree/develop/examples/post_training_quantization/openvino/quantize_with_accuracy_control) that demonstrates how to quantize [Student-Teacher Feature Pyramid Matching (STFPM)](https://huggingface.co/alexsu52/stfpm_mvtec_capsule) OpenVINO model from [Anomalib](https://github.com/openvinotoolkit/anomalib). I hope this example helps you with fast start.

More details about post-training quantization with NNCF [here](https://docs.openvino.ai/latest/nncf_ptq_introduction.html)

**Comment by alexsu52** [comment]
Hi @glucasol,

Taking into account the model compression rate 1.057 from your log, I can assume that only some layers of the model have been quantized. Could you share the original model or the name of the anomalib model you used? Also, could you share full log of your script run?

Thanks!

**Comment by glucasol** [comment]
Hi @alexsu52 ,

I tested other model and different accuracy drops to see if the results were different. Unfortunately I cannot share the model because it is too large, but you can replicate it just following the steps on [Anomalib's repo](https://github.com/openvinotoolkit/anomalib). Just install the dependencies and before run, check the [config.yaml](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/models/padim/config.yaml) file and ensure that export_mode is set to "openvino", like this:
```
optimization:
  export_mode: "openvino" # options: openvino, onnx
```
After that, just run the command:
```python tools/train.py ```

And the openvino IR model will be generated.

For the tests I have done, using the same model I changed the accuracy drop from 0.0005 to 0.0015 and 0.005, but all situation returns a **Model compression rate:  3.985** and **Performance speed up (throughput mode)** almost equal. As far as I increase accuracy drop, It was supposed to increase the model compression rate and performance speed up right? But this doesn't happen, as you can see the log files bellow.

[log_mvtec0005.txt](https://github.com/openvinotoolkit/openvino/files/11554620/log_mvtec0005.txt)
[log_mvtec0015.txt](https://github.com/openvinotoolkit/openvino/files/11554621/log_mvtec0015.txt)
[log_mvtec_005.txt](https://github.com/openvinotoolkit/openvino/files/11554622/log_mvtec_005.txt)


Thanks for your time!

**Comment by alexsu52** [comment]
Hi @glucasol,

I could not reproduce the issue with padim model from the [Anomalib's repo](https://github.com/openvinotoolkit/anomalib) using nncf 2.4.0. But NNCF from the develop branch has issues and I prepared the fix that you can try https://github.com/openvinotoolkit/nncf/pull/1902. But your issue, probably, has another root. Could you run benchmark_app for quantized model to collect execution report? Also, please share your hardware configuration, OpenVINO version. 

The command to collect the execution report is following:
```
benchmark_app -m <path to the quantized model> -report_type average_counters
```
benchmark_app will generate `benchmark_average_counters_report.csv`. Please share with me.

I highly recommend to use the latest OpenVINO version from pypi (2023.0.0).

**Comment by glucasol** [comment]
Hi @alexsu52 ,
These are my configuration:

- Intel(R) Xeon(R) Gold 6150 CPU @ 2.70GHz
- 700GB RAM
- Ubuntu 20.04.4 LTS

I have cloned the branch you suggested, uploaded the OpenVINO version to latest and tested again. Below are the benchmark .csv files.

[benchmark_average_counters_report.csv](https://github.com/openvinotoolkit/openvino/files/11678565/benchmark_average_counters_report.csv)
[benchmark_report.csv](https://github.com/openvinotoolkit/openvino/files/11678567/benchmark_report.csv)


Thanks!

**Comment by dmitry-gorokhov** [comment]
Hi @glucasol, Intel(R) Xeon(R) Gold 6150 doesn't support Intel(R) DL boost (aka VNNI). Such technology was firstly introduced in second generation of Xeon processors (code name Cascad Lake). Older generation have very limited HW capabilities for efficient 8bit inference. 
Intel(R) Core(TM) i9-10980XE has DL boost support that's why numbers provided by @alexsu52 shows 2x performance boost for quantized model.

**Comment by avitial** [comment]
Closing this, I hope previous responses were sufficient to help you proceed. Feel free to reopen to ask any questions related to this topic.
